GeneralsMD/Code/GameEngine/Source/Common/RTS/Money.cpp

```diff
@@ -100,6 +102,46 @@ void Money::deposit(UnsignedInt amountToDeposit, Bool playSound)
 	}
 }
 
+// ------------------------------------------------------------------------------------------------
+void Money::setStartingCash(UnsignedInt amount)
+{
+	m_startingCash = amount;
+	m_money = amount;
+	m_currentBucket = 0;
+	m_lastBucketFrame = 0;
+	for (UnsignedInt i = 0; i < 60; ++i)
+		m_incomeBuckets[i] = 0;
+}
+
+// ------------------------------------------------------------------------------------------------
+void Money::updateIncomeBucket()
+{
+	UnsignedInt frame = TheGameLogic->getFrame();
+	UnsignedInt lastSec = m_lastBucketFrame / LOGICFRAMES_PER_SECOND;
```

Comment:
Can this value not be derived from `m_currentBucket` ?

Reply:
Currently, the function is overengineered. I suggest try throw out every thing that is not needed and support saveload through xfer instead.

The tiniest working implementation for this function is:

```cpp
void Money::updateIncomeBucket()
{
	UnsignedInt frame = TheGameLogic->getFrame();
	UnsignedInt nextBucket = (frame / LOGICFRAMES_PER_SECOND) % ARRAY_SIZE(m_incomeBuckets);
	if (nextBucket != m_currentBucket)
	{
		m_currentBucket = nextBucket;
		m_incomeBuckets[m_currentBucket] = 0u;
	}
}
```